apps/elixir_ls_debugger/lib/debugger/server.ex

```diff
@@ -1297,4 +1280,63 @@ defmodule ElixirLS.Debugger.Server do
     server = Process.info(self())[:registered_name] || self()
     {__MODULE__, reason, [server]}
   end
+
+  defp update_threads(state = %__MODULE__{}) do
+    pids = :erlang.processes()
+    {state, thread_ids, new_ids} = ensure_thread_ids(state, pids)
+
+    for thread_id <- new_ids do
+      Output.send_event("thread", %{
+        "reason" => "started",
+        "threadId" => thread_id
+      })
+    end
+
+    exited_pids = Map.keys(state.threads_inverse) -- pids
+
+    state =
+      Enum.reduce(exited_pids, state, fn pid, state ->
+        handle_process_exit(state, pid)
+      end)
+
+    {schedule_update_threads(state), thread_ids}
+  end
+
+  defp handle_process_exit(state = %__MODULE__{}, pid) when is_pid(pid) do
+    {thread_id, threads_inverse} = state.threads_inverse |> Map.pop(pid)
+    paused_processes = remove_paused_process(state, pid)
+
+    state = %__MODULE__{
+      state
+      | threads: state.threads |> Map.delete(thread_id),
```

Comment:
the pipeline here isn't helping readabiliy.
```elixir
threads: Map.delete(state.threads, thread_id)
```
is much clearer to my eyes.

Reply:
I can change it. I'm just not that strict on that rule. IMO if there's a clear data transformation then one call pipe is OK. It also reads similar to method calls from OOP languages

Reply:
Thank you, 
I've noticed this pattern a lot; really pipelines are for a series of data transformations, so we don't have to do that whole `map0`, `map1` thing we see so much in erlang, otherwise it's just noise. 

Also, we should definitely not try to replicate what we see in OOP languages ;) Elixir lets us climb out of that tarpit.